### PR TITLE
feat: upgrade to Angular 16, mark components as standalone

### PR DESCRIPTION
**Pull request by arturovt**
This commit upgrades the workspace to Angular 16, as it includes security fixes necessary to ensure compatibility with content security policies (this also unblocks me from proceeding with other issues). All components have been designated as `standalone`, eliminating the need for them to be declared within the `HotToastModule`. Consequently, `HotToastModule` is now deprecated, as it is unnecessary for standalone applications.

### CODE REVIEW

**Comment by arturovt** [comment]
@shhdharmen friendly ping.

**Comment by arturovt** [comment]
@NetanelBasal friendly ping.

**Comment by shhdharmen** [comment]
I will review it this weekend.

File: projects/ngneat/hot-toast/schematics/ng-add/utils/index.ts [hunk 1]
```diff
@@ -48,7 +47,7 @@ export function addStyleToTarget(
 
 export function getProjectFromWorkspace(workspace: WorkspaceSchema, projectName?: string): WorkspaceProject {
   /* eslint-disable-next-line @typescript-eslint/no-non-null-assertion */
-  const project = workspace.projects[projectName || workspace.defaultProject!];
+  const project = workspace.projects[projectName!];
```
Comment (shhdharmen) [inline projects/ngneat/hot-toast/schematics/ng-add/utils/index.ts:50]: Why removed default project?

File: projects/ngneat/hot-toast/src/lib/hot-toast.module.ts [hunk 2]
```diff
@@ -1,32 +1,11 @@
...
-    CheckMarkComponent,
-    ErrorComponent,
-    LoaderComponent,
-    WarningComponent,
-    InfoComponent,
-  ],
-  imports: [CommonModule, DynamicViewModule],
-})
+/**
+ * @deprecated Use `provideHotToastConfig` to provide a global toast configuration.
```
Comment (shhdharmen) [inline projects/ngneat/hot-toast/src/lib/hot-toast.module.ts:6]: We don't want to deprecate usage of modules.

**Review by shhdharmen — CHANGES_REQUESTED** [review]
- Convert docs application to standalone
- Make sure to have cypress test cases for standard and standalone, both
- Add example in docs app to showcase how to use the library with standalone component
- Change target branch to beta

**Pull request merged**